### ISSUE TITLE
Site Editor: "Site" menu item is active when editing a template

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#36821]
<img width="1096" alt="Screenshot 2021-11-24 at 14 35 38" src="https://user-images.githubusercontent.com/846565/143258083-cf438aca-7bf4-46c3-a5eb-c04224826ef0.png">

The "Site" menu item should not be active in this case.

**Comment by Mamaduka** [comment]
Can you give me more details/specs when this menu item should be active and when not?

**Comment by jameskoster** [comment]
Sure. Essentially, the "Site" menu item should only be active when:

* You click "Appearance > Editor" (this works currently)
* You click the Site menu item itself (this also works)

In any other case the flow is template-centric (IE you are navigating through the Templates List), so it doesn't make sense to make the "Site" item active in those situations.

I appreciate this is a little tricky to pin down since template + site editing exist in the same package, but the split in the UX makes this feel quite important.

**Comment by kevin940726** [comment]
The problem is that we do a redirection upon clicking on the "Site" link. There's essentially no page which is called "Site", we immediately redirect the user to the template editor (which is the frontpage template by default). Should we:

- Highlight the "Site" link only when we're on the default(frontpage) template, or
- Highlight the "Site" link whenever we're editing any template but not template parts,
- Or anything else

Since we always do a full navigation, so we don't have the information of where we're coming from (whether we click on the "Site" link or not). Since this would likely be linked to the `aria-current` attribute in https://github.com/WordPress/gutenberg/pull/36946, I guess it makes sense to have accessibility folks giving feedbacks on this as well. WDYT?

**Comment by jameskoster** [comment]
None of these options seem very good to me. It would be very strange imo to see "Site" as the active menu item when editing the Page template via the templates list (assuming a static home page it set in this example).

Probably a dumb question but... logic exists somewhere to interpret the homepage settings in order to display the correct template / content when opening the site editor. Can we not use that same logic to determine when to set the active state on the Site menu item?

**Comment by Mamaduka** [comment]
I think we can check if `page?.path === siteUrl`. Currently, this is only true for the home/index page.

**Comment by jameskoster** [comment]
@Mamaduka would that same logic allow us to display the site name / url in the top bar when editing the homepage? (https://github.com/WordPress/gutenberg/issues/36207)

**Comment by Mamaduka** [comment]
Technically yes. I can create a proof of concept PR later this week.

**Comment by kevin940726** [comment]
> Probably a dumb question but... logic exists somewhere to interpret the homepage settings in order to display the correct template / content when opening the site editor. Can we not use that same logic to determine when to set the active state on the Site menu item?

Isn't this the same as my first proposal? Or am I missing something 🤔 ?

> Highlight the "Site" link only when we're on the default(frontpage) template

**Comment by jameskoster** [comment]
I don't think it's quite the same, but perhaps I am misinterpreting. 

> Highlight the "Site" link only when we're on the default(frontpage) template

In this case, if a static page were set as the front page, wouldn't this mean that the "Site" menu item would be active if you navigated to Templates > Page? That would be kind of confusing.

**Comment by Mamaduka** [comment]
@jameskoster, I tried to make this work, but it requires more changes than I expected.

I'm going to remove this from the WP 5.9 board since it's not a block, and we're already in the RC phase.

**Comment by carolinan** [comment]
By now, after having used the Site Editor for some time, I have a habit of using the Site link a lot and I would prefer it it was not removed.

A regular flow can look like:
Open Site Editor. The index/home/front page template is shown.
I open the navigation sidebar.
Select Templates
Select the template I want to work on from the list.

Now when I have made the changes and saved,  if I want to go back to the home, I open the navigation sidebar and select "Site".

I never open the navigation sidebar, select templates, and select home from there. Why would I? It is more clicks...
